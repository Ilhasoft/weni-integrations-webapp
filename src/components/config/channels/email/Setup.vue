<template>
  <unnnic-modal class="gmail-setup" :close-icon="false" @click.stop :text="$t(`gmail.setup.title`)">
    <template #icon> <img class="logo" src="../../../../assets/svgs/gmail.svg" alt="" /></template>
    <template #message>
      <div>
        <span v-html="$t(`gmail.setup.description`)"></span>
      </div>
    </template>

    <template #options>
      <div class="gmail-setup__buttons">
        <unnnic-button
          class="gmail-setup__buttons__cancel"
          type="tertiary"
          size="large"
          :text="$t('general.Cancel')"
          @click="closePopUp"
        />

        <unnnic-button
          class="gmail-setup__buttons__continue"
          size="large"
          :text="$t('gmail.setup.buttons.continue')"
          @click="login"
        />
      </div>
    </template>
  </unnnic-modal>
</template>

<script>
  import { googleSdkLoaded } from 'vue3-google-login';
  import getEnv from '../../../..//utils/env';
  import { mapState } from 'pinia';
  import { auth_store } from '@/stores/modules/auth.store';
  export default {
    name: 'gmailSetup',
    data() {
      return {
        gmailCallback: (response) => console.log(response),
      };
    },
    computed: {
      ...mapState(auth_store, ['project']),
    },
    methods: {
      closePopUp() {
        this.$emit('closePopUp');
      },
      login() {
        console.log('aloo');
        googleSdkLoaded((google) => {
          google.accounts.oauth2
            .initCodeClient({
              client_id: getEnv('VITE_APP_GOOGLE_CLOUD_ID'),
              scope: 'email profile openid',
              redirect_uri: 'https://dash.stg.cloud.weni.ai/auth/callback',
              callback: (response) => {
                console.log(response);
              },
            })
            .requestCode();
        });
      },
    },
  };
</script>
<style lang="scss" scoped>
  .gmail-setup {
    &__buttons {
      display: grid;
      grid-template-columns: 1fr 1fr;
      justify-content: space-around;
      gap: $unnnic-spacing-inline-xs;

      &__cancel,
      &__continue {
        width: 100%;
      }
    }
  }
</style>
