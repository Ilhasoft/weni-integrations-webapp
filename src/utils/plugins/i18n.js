import Vue from 'vue';
import VueI18n from 'vue-i18n';

Vue.use(VueI18n);

import en from '../../locales/en.json';
import pt_br from '../../locales/pt_br.json';
import es_es from '../../locales/es_es.json';

const languages = {
  'en-us': en,
  'pt-br': pt_br,
  es: es_es,
};

const messages = Object.assign(languages);

const i18n = new VueI18n({
  locale: 'en-us',
  fallbackLocale: 'en-us',
  messages,
});

export default i18n;
