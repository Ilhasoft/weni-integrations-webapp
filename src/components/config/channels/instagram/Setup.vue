<template>
  <div>
    <unnnic-modal
      v-if="stage === 'login'"
      ref="instagram-setup-modal"
      class="instagram-setup"
      :text="$t('Instagram.setup.title')"
      modal-icon="social-instagram-1"
      :close-icon="false"
      @close="closePopUp"
      @click.stop
    >
      <div slot="message">
        <span v-html="$t('Instagram.setup.description')"></span>
      </div>
      <div slot="options">
        <div class="instagram-setup__buttons">
          <unnnic-button
            class="instagram-setup__buttons__cancel"
            type="terciary"
            size="large"
            :text="$t('general.Cancel')"
            @click="closePopUp"
          ></unnnic-button>

          <LoadingButton
            class="instagram-setup__buttons__start"
            type="secondary"
            size="large"
            :text="$t('WhatsAppCloud.setup.continue')"
            :isLoading="onLogin"
            :disabled="onLogin"
            @clicked="startFacebookLogin"
          />
        </div>
      </div>
    </unnnic-modal>

    <unnnic-modal
      v-else
      ref="page-selection-modal"
      class="page-selection"
      :text="$t('Instagram.setup.title')"
      modal-icon="social-instagram-1"
      :close-icon="false"
      @close="closePopUp"
      @click.stop
    >
      <div slot="message" class="page-selection__select">
        <span v-html="$t('Instagram.setup.page_selection.description')"></span>
        <unnnic-select
          ref="page-selection-input"
          :search="false"
          size="sm"
          :value="selectedPage"
          @input="handlePageSelection"
          :key="selectKey"
        >
          <option
            v-for="(page, index) in pageList"
            :key="index"
            :value="page.id"
            :label="page.name"
          >
            {{ page.name }}
          </option>
        </unnnic-select>
      </div>

      <div class="page-selection__buttons" slot="options">
        <unnnic-button
          class="page-selection__buttons__cancel"
          type="terciary"
          size="large"
          :text="$t('WhatsAppCloud.config.phone_numbers.connect_later')"
          @click="closePopUp"
        ></unnnic-button>

        <LoadingButton
          class="page-selection__buttons__save"
          type="secondary"
          size="large"
          :isLoading="loadingUpdateAppConfig || loadingCreateApp"
          :loadingText="$t('general.loading')"
          :text="$t('WhatsAppCloud.config.phone_numbers.create_channel')"
          @clicked="createChannel"
        />
      </div>
    </unnnic-modal>
  </div>
</template>

<script>
  import axios from 'axios';
  import { mapActions, mapState } from 'vuex';
  import { unnnicCallAlert } from '@weni/unnnic-system';
  import LoadingButton from '../../../LoadingButton.vue';
  import getEnv from '../../../..//utils/env';
  import { initFacebookSdk } from '../../../../utils/plugins/fb';

  export default {
    name: 'InstagramSetup',
    components: {
      LoadingButton,
    },
    props: {
      app: {
        type: Object,
        default: /* istanbul ignore next */ () => {},
      },
    },
    data() {
      return {
        stage: 'login',
        accessToken: null,
        pageList: [],
        selectedPage: null,
        loadingInstagramConfigure: false,
        selectKey: 0,
        onLogin: false,
      };
    },
    async mounted() {
      window.startPageSelectionStage = this.startPageSelectionStage;
      window.changeLoginState = this.changeLoginState;
    },
    computed: {
      ...mapState({
        project: (state) => state.auth.project,
        createAppResponse: (state) => state.appType.createAppResponse,
        loadingCreateApp: (state) => state.appType.loadingCreateApp,
        errorCreateApp: (state) => state.appType.errorCreateApp,
        loadingUpdateAppConfig: (state) => state.appType.loadingUpdateAppConfig,
        errorUpdateAppConfig: (state) => state.appType.errorUpdateAppConfig,
      }),
    },
    methods: {
      ...mapActions(['createApp', 'updateAppConfig']),
      handlePageSelection(page) {
        this.selectedPage = page;
        this.selectKey += 1;
      },
      changeLoginState(state) {
        this.onLogin = state;
      },
      /* istanbul ignore next */
      async startFacebookLogin() {
        const fbAppId = getEnv('VUE_APP_FACEBOOK_APP_ID');

        if (!fbAppId) {
          return;
        }

        /* eslint-disable no-undef */
        const loginCallback = async () => {
          this.changeLoginState(true);
          FB.login(
            async function (response) {
              if (response.authResponse && response.authResponse.grantedScopes) {
                const accessToken = response.authResponse.accessToken;
                this.startPageSelectionStage(accessToken);
              }
              this.changeLoginState(false);
            },
            {
              return_scopes: true,
              scope:
                'instagram_basic,instagram_manage_messages,pages_manage_metadata,pages_messaging,pages_read_engagement,pages_show_list',
            },
          );
        };

        initFacebookSdk(fbAppId, loginCallback);
      },
      /* istanbul ignore next */
      async startPageSelectionStage(accessToken) {
        this.accessToken = accessToken;
        this.stage = 'select-page';

        this.loadingPages = true;
        try {
          const fbAccountUrl = `https://graph.facebook.com/me/accounts?access_token=${accessToken}&limit=200`;
          const res = await axios.get(fbAccountUrl);
          this.pageList = res.data.data;
        } catch (error) {
          this.callModal({ type: 'Error', text: this.$t('Instagram.setup.account_data.error') });
          return;
        }

        this.loadingPages = false;
      },
      async createChannel() {
        const page = this.pageList.find((page) => page.id === this.selectedPage);

        if (!page) {
          this.callModal({ type: 'Error', text: this.$t('Instagram.setup.find_page.error') });
          return;
        }

        await this.createApp({ code: this.app.code, payload: { project_uuid: this.project } });
        if (this.errorCreateApp) {
          this.callModal({ type: 'Error', text: this.$t('Instagram.setup.create_app.error') });
          return;
        }

        const data = {
          code: this.app.code,
          appUuid: this.createAppResponse.uuid,
          payload: {
            config: {
              user_access_token: page.access_token,
              page_name: page.name,
              page_id: page.id,
              fb_user_id: FB.getUserID(),
            },
            channeltype_code: 'IG',
            project_uuid: this.project,
          },
        };

        await this.updateAppConfig(data);

        if (this.errorUpdateAppConfig) {
          this.callModal({ type: 'Error', text: this.$t('Instagram.setup.update_app.error') });
          return;
        }

        this.callModal({ type: 'Success', text: this.$t('Instagram.setup.success') });
        this.$router.replace('/apps/my');
      },
      closePopUp() {
        this.$emit('closePopUp');
      },
      callModal({ text, type }) {
        unnnicCallAlert({
          props: {
            text: text,
            title: type,
            icon: type === 'Success' ? 'check-circle-1-1' : 'alert-circle-1',
            scheme: type === 'Success' ? 'feedback-green' : 'feedback-red',
            position: 'bottom-right',
            closeText: this.$t('general.Close'),
          },
          seconds: 6,
        });
      },
    },
  };
</script>

<style lang="scss" scoped>
  .instagram-setup {
    cursor: default;

    &__buttons {
      display: flex;
      justify-content: space-around;
      gap: $unnnic-spacing-inline-xs;

      &__cancel,
      &__start {
        width: 50%;
      }
    }

    ::v-deep {
      .unnnic-modal-container-background-body-description {
        padding-bottom: $unnnic-spacing-stack-nano;
      }
      .link {
        color: inherit;
        text-decoration: none;
        border-bottom: $unnnic-border-width-thinner solid $unnnic-color-neutral-cloudy;
      }
    }
  }

  .page-selection {
    cursor: default;

    &__buttons {
      display: flex;
      justify-content: space-around;
      gap: $unnnic-spacing-inline-xs;

      &__cancel,
      &__save {
        width: 50%;
      }
    }

    &__select {
      display: flex;
      flex-direction: column;
      gap: $unnnic-spacing-stack-md;
    }

    ::v-deep .unnnic-modal-container-background-body-description {
      padding-bottom: $unnnic-spacing-stack-nano;
    }

    ::v-deep .unnnic-modal-container-background-button {
      padding-top: $unnnic-spacing-stack-md;
    }
  }
</style>
