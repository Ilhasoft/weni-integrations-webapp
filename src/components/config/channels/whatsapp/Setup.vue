<template>
  <div>
    <unnnic-modal
      ref="whatsapp-setup-modal"
      class="whatsapp-setup"
      :text="$t('WhatsAppCloud.setup.connect')"
      :description="$t('WhatsAppCloud.setup.description')"
      scheme="feedback-green"
      modal-icon="phone-3"
      :close-icon="false"
      @close="closePopUp"
    >
      <template #options>
        <div>
          <div class="whatsapp-setup__buttons">
            <unnnic-button
              class="whatsapp-setup__buttons__cancel"
              type="tertiary"
              size="large"
              :text="$t('general.Cancel')"
              @click="closePopUp"
            ></unnnic-button>

            <LoadingButton
              class="whatsapp-setup__buttons__start"
              type="secondary"
              size="large"
              :text="$t('WhatsAppCloud.setup.continue')"
              :isLoading="onLogin"
              :disabled="onLogin"
              @clicked="startFacebookLogin"
            />
          </div>
        </div>
      </template>
    </unnnic-modal>
  </div>
</template>

<script>
  import { mapActions, mapState } from 'pinia';
  import { whatsapp_cloud } from '@/stores/modules/appType/channels/whatsapp_cloud.store';
  import { auth_store } from '@/stores/modules/auth.store';
  import alert from '@/utils/call';
  import LoadingButton from '../../../LoadingButton/index.vue';
  import getEnv from '../../../..//utils/env';
  import { initFacebookSdk } from '../../../../utils/plugins/fb';

  export default {
    name: 'WhatsAppSetup',
    components: {
      LoadingButton,
    },
    props: {
      app: {
        type: Object,
        default: /* istanbul ignore next */ () => {},
      },
    },
    data() {
      return {
        phoneNumberId: null,
        wabaId: null,
        onLogin: false,
      };
    },
    async mounted() {
      window.createChannel = this.createChannel;
      window.changeLoginState = this.changeLoginState;
    },
    computed: {
      ...mapState(auth_store, ['project']),
      ...mapState(whatsapp_cloud, ['loadingWhatsAppCloudConfigure', 'errorCloudConfigure']),
    },
    methods: {
      ...mapActions(whatsapp_cloud, ['configurePhoneNumber']),
      changeLoginState(state) {
        this.onLogin = state;
      },
      /* istanbul ignore next */
      async startFacebookLogin() {
        const fbAppId = getEnv('VUE_APP_WHATSAPP_FACEBOOK_APP_ID');
        const configId = getEnv('VUE_APP_WHATSAPP_FACEBOOK_APP_CONFIG_ID');

        if (!fbAppId) {
          return;
        }

        /* eslint-disable no-undef */
        const loginCallback = async () => {
          this.changeLoginState(true);

          const sessionInfoListener = (event) => {
            if (event.origin !== 'https://www.facebook.com') return;
            try {
              const data = JSON.parse(event.data);
              if (data.type === 'WA_EMBEDDED_SIGNUP') {
                // if user finishes the Embedded Signup flow
                if (data.event === 'FINISH') {
                  const { phone_number_id, waba_id } = data.data;
                  this.phoneNumberId = phone_number_id;
                  this.wabaId = waba_id;
                }
                // if user cancels the Embedded Signup flow
                else {
                  const { current_step } = data.data;
                  console.log(
                    'Session info listener: User cancelled login or did not fully authorize.',
                    current_step,
                  );
                }
              }
            } catch {
              // Don’t parse info that’s not a JSON
              console.log('Non JSON Response', event.data);
            }
          };

          window.addEventListener('message', sessionInfoListener);

          typeof fbq !== 'undefined' &&
            fbq('trackCustom', 'WhatsAppOnboardingStart', {
              appId: fbAppId,
              feature: 'whatsapp_embedded_signup',
            });

          FB.login(
            async function (response) {
              if (response.authResponse) {
                const code = response.authResponse.code;
                await this.createChannel(code);
              } else {
                console.log('User cancelled login or did not fully authorize.');
              }
              this.changeLoginState(false);
            },
            {
              config_id: configId,
              response_type: 'code',
              override_default_response_type: true,
              extras: {
                sessionInfoVersion: 2,
              },
            },
          );
        };

        initFacebookSdk(fbAppId, loginCallback);
      },
      async createChannel(code) {
        if (!this.loadingWhatsAppCloudConfigure) {
          const data = {
            waba_id: this.wabaId,
            phone_number_id: this.phoneNumberId,
            project_uuid: this.project,
            auth_code: code,
          };

          await this.configurePhoneNumber({ data });

          if (this.errorCloudConfigure) {
            this.callErrorModal({
              text: this.$t('WhatsAppCloud.config.configure_phone_number.error'),
            });
          } else {
            this.$router.replace('/apps/my');
          }
        }
      },
      closePopUp() {
        this.$emit('closePopUp');
      },
      callErrorModal({ text }) {
        alert.callAlert({
          props: {
            text: text,
            type: 'error',
          },
          seconds: 6,
        });
      },
    },
  };
</script>

<style lang="scss" scoped>
  .whatsapp-setup {
    cursor: default;

    &__buttons {
      display: flex;
      justify-content: space-around;
      gap: $unnnic-spacing-inline-xs;

      &__cancel,
      &__start {
        width: 50%;
      }
    }

    ::v-deep .unnnic-modal-container-background-body-description {
      padding-bottom: $unnnic-spacing-stack-lg;
    }
  }
</style>
