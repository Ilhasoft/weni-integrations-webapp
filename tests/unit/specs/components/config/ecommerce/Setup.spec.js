import { unnnicCallAlert as mockUnnnicCallAlert } from '@weni/unnnic-system';

jest.mock('@weni/unnnic-system', () => ({
  ...jest.requireActual('@weni/unnnic-system'),
  unnnicCallAlert: jest.fn(),
}));

import Vuex from 'vuex';
import { mount, createLocalVue } from '@vue/test-utils';
import Setup from '@/components/config/ecommerce/vtex/Setup.vue';
import i18n from '@/utils/plugins/i18n';
import { singleApp } from '../../../../../__mocks__/appMock';
import '@weni/unnnic-system';

const localVue = createLocalVue();
localVue.use(Vuex);

// Global mock
navigator.clipboard = {
  writeText: jest.fn(),
};

const mountComponent = async ({
  errorConfiguredApps = false,
  errorCreateApp = false,
  multipleConfigured = false,
} = {}) => {
  let configuredApps = [
    {
      ...singleApp,
      code: 'wpp-cloud',
      config: { wa_certified_name: 'wpp name', title: 'wpp title' },
    },
  ];

  if (multipleConfigured) {
    configuredApps.push({
      ...singleApp,
      code: 'wpp-cloud',
      config: { wa_certified_name: 'wpp name', title: 'wpp title' },
    });
  }

  const state = {
    auth: {
      project: '123',
    },
    myApps: {
      configuredApps,
      loadingConfiguredApps: false,
      errorConfiguredApps,
    },
    appType: {
      loadingCreateApp: false,
      errorCreateApp,
    },
  };

  const actions = {
    createApp: jest.fn(),
    getConfiguredApps: jest.fn(),
  };

  const store = new Vuex.Store({
    modules: {
      ecommerce: {
        namespaced: true,
        actions: {
          getVtexAppUuid: jest.fn(),
        },
        state: {
          generatedVtexAppUuid: 'generated123',
          errorVtexAppUuid: null,
        },
      },
    },
    state,
    actions,
  });

  const wrapper = mount(Setup, {
    localVue,
    i18n,
    store,
    propsData: {
      app: {
        ...singleApp,
        code: 'vtex',
      },
    },
  });

  await wrapper.vm.$nextTick();

  return { wrapper, state, actions };
};

describe('components/config/ecommerce/vtex/Setup.vue', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should be rendered properly', async () => {
    const { wrapper } = await mountComponent();
    expect(wrapper).toMatchSnapshot();
  });

  it('should call error modal if failing to get configured apps', async () => {
    const { wrapper } = await mountComponent({ errorConfiguredApps: true });

    await wrapper.vm.$nextTick();

    expect(mockUnnnicCallAlert).toHaveBeenCalledWith({
      props: {
        text: 'We were not able to fetch your configured apps, please try again later.',
        title: 'Error',
        icon: 'alert-circle-1',
        scheme: 'feedback-red',
        closeText: 'Close',
        position: 'bottom-right',
      },
      seconds: 6,
    });

    expect(wrapper.emitted('closePopUp')).toBeTruthy();
  });

  it('should close popup', async () => {
    const { wrapper } = await mountComponent({ multipleConfigured: true });

    const closeButton = wrapper.findComponent({ ref: 'unnnic-vtex-modal-close-button' });
    await closeButton.trigger('click');

    expect(wrapper.emitted('closePopUp')).toBeTruthy();
  });

  it('should call error modal if any of the fields are empty', async () => {
    const { wrapper } = await mountComponent();

    const setupButton = wrapper.findComponent({ ref: 'unnnic-vtex-modal-setup-button' });
    await setupButton.trigger('click');

    expect(mockUnnnicCallAlert).toHaveBeenCalledWith({
      props: {
        text: 'All fields are mandatory',
        title: 'Error',
        icon: 'alert-circle-1',
        scheme: 'feedback-red',
        closeText: 'Close',
        position: 'bottom-right',
      },
      seconds: 6,
    });
  });

  it('should setup vtex with correct information', async () => {
    const { wrapper, actions } = await mountComponent();

    wrapper.vm.selectedChannel = [{ value: '456' }];
    wrapper.vm.storeDomain = 'storedomain.com';
    wrapper.vm.apiDomain = 'apidomain.com';
    wrapper.vm.appKey = 'appKey123';
    wrapper.vm.appToken = 'appToken123';

    const setupButton = wrapper.findComponent({ ref: 'unnnic-vtex-modal-setup-button' });
    // finish step 1
    await setupButton.trigger('click');

    // click again to finish step 2
    await setupButton.trigger('click');

    await wrapper.vm.$nextTick();

    expect(actions.createApp).toHaveBeenCalledWith(expect.anything(), {
      code: 'vtex',
      payload: {
        project_uuid: '123',
        domain: 'apidomain.com',
        store_domain: 'storedomain.com',
        app_key: 'appKey123',
        app_token: 'appToken123',
        wpp_cloud_uuid: '456',
        uuid: 'generated123',
      },
    });
  });

  it('should call error modal with invalid credentials information on create error', async () => {
    const { wrapper } = await mountComponent({
      errorCreateApp: {
        response: { status: 400, data: { detail: 'The credentials provided are invalid.' } },
      },
    });

    wrapper.vm.selectedChannel = [{ value: '456' }];
    wrapper.vm.storeDomain = 'storedomain.com';
    wrapper.vm.apiDomain = 'apidomain.com';
    wrapper.vm.appKey = 'appKey123';
    wrapper.vm.appToken = 'appToken123';

    const setupButton = wrapper.findComponent({ ref: 'unnnic-vtex-modal-setup-button' });
    // finish step 1
    await setupButton.trigger('click');

    // click again to finish step 2
    await setupButton.trigger('click');

    await wrapper.vm.$nextTick();

    expect(mockUnnnicCallAlert).toHaveBeenCalledWith({
      props: {
        text: 'Invalid credentials for the given subdomain',
        title: 'Error',
        icon: 'alert-circle-1',
        scheme: 'feedback-red',
        closeText: 'Close',
        position: 'bottom-right',
      },
      seconds: 6,
    });
  });

  it('should call default error on unknown create error', async () => {
    const { wrapper } = await mountComponent({
      errorCreateApp: {
        response: { status: 400, data: { detail: 'Unknown error' } },
      },
    });

    wrapper.vm.selectedChannel = [{ value: '456' }];
    wrapper.vm.storeDomain = 'storedomain.com';
    wrapper.vm.apiDomain = 'apidomain.com';
    wrapper.vm.appKey = 'appKey123';
    wrapper.vm.appToken = 'appToken123';

    const setupButton = wrapper.findComponent({ ref: 'unnnic-vtex-modal-setup-button' });
    // finish step 1
    await setupButton.trigger('click');

    // click again to finish step 2
    await setupButton.trigger('click');

    await wrapper.vm.$nextTick();

    expect(mockUnnnicCallAlert).toHaveBeenCalledWith({
      props: {
        text: 'An error occurred while trying to integrate with VTEX, please try again later',
        title: 'Error',
        icon: 'alert-circle-1',
        scheme: 'feedback-red',
        closeText: 'Close',
        position: 'bottom-right',
      },
      seconds: 6,
    });
  });

  it('should save to clipboard on copy and call success modal', async () => {
    const { wrapper } = await mountComponent();

    wrapper.vm.selectedChannel = [{ value: '456' }];
    wrapper.vm.storeDomain = 'storedomain.com';
    wrapper.vm.apiDomain = 'apidomain.com';
    wrapper.vm.appKey = 'appKey123';
    wrapper.vm.appToken = 'appToken123';

    const setupButton = wrapper.findComponent({ ref: 'unnnic-vtex-modal-setup-button' });
    // finish step 1
    await setupButton.trigger('click');

    await wrapper.vm.$nextTick();

    const copyButton = wrapper.findComponent({ ref: 'vtex-copy-button' });
    await copyButton.trigger('click');

    expect(navigator.clipboard.writeText).toHaveBeenCalledWith(wrapper.vm.webhookUrl);
    expect(mockUnnnicCallAlert).toHaveBeenCalledWith({
      props: {
        text: 'Successfully copied!',
        title: 'Success',
        icon: 'check-circle-1-1',
        scheme: 'feedback-green',
        closeText: 'Close',
        position: 'bottom-right',
      },
      seconds: 3,
    });
  });

  describe('callModal', () => {
    beforeEach(() => {
      jest.clearAllMocks();
    });

    it('should call unnnicCallAlert with Success state', async () => {
      const { wrapper } = await mountComponent();

      expect(mockUnnnicCallAlert).not.toHaveBeenCalled();

      wrapper.vm.callModal({ text: 'success', type: 'Success' });

      expect(mockUnnnicCallAlert).toHaveBeenCalledWith({
        props: {
          text: 'success',
          title: 'Success',
          icon: 'check-circle-1-1',
          scheme: 'feedback-green',
          closeText: 'Close',
          position: 'bottom-right',
        },
        seconds: 6,
      });
    });

    it('should call unnnicCallAlert with Error state', async () => {
      const { wrapper } = await mountComponent();

      expect(mockUnnnicCallAlert).not.toHaveBeenCalled();

      wrapper.vm.callModal({ text: 'error', type: 'Error' });

      expect(mockUnnnicCallAlert).toHaveBeenCalledWith({
        props: {
          text: 'error',
          title: 'Error',
          icon: 'alert-circle-1',
          scheme: 'feedback-red',
          closeText: 'Close',
          position: 'bottom-right',
        },
        seconds: 6,
      });
    });
  });
});
