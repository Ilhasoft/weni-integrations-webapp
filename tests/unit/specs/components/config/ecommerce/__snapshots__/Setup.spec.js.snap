// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/config/ecommerce/vtex/Setup.vue should be rendered properly 1`] = `
<transition-stub data-v-4c459454="" name="modal" class="vtex-modal">
  <div data-v-4c459454="" class="unnnic-modal">
    <div data-v-4c459454="" class="unnnic-modal-container">
      <div data-v-4c459454="" class="unnnic-modal-container-background">
        <div data-v-4c459454="" class="unnnic-modal-container-background-body">
          <!---->
          <!---->
          <!---->
          <div data-v-4c459454="" class="unnnic-modal-container-background-body-title"> Integrate with VTEX </div>
        </div>
        <div data-v-4c459454="" class="unnnic-modal-container-background-body-description-container">
          <div data-v-4c459454="" class="unnnic-modal-container-background-body-description"> To proceed with the instalation insert a subdomain <div class="vtex-modal__content">
              <div class="vtex-modal__content__form">
                <div>
                  <p class="unnnic-label__label">WhatsApp Number</p>
                  <div class="unnnic-select-smart"><span data-v-6fe2671f="" class="dropdown"><div data-v-400546b3="" class="unnnic-select-smart__input text-input size--md has-icon-right" readonly="readonly" data-v-6fe2671f=""><input data-v-97b44e8c="" data-v-400546b3="" type="text" size="23" readonly="readonly" placeholder="undefined - (wpp title)" iconleft="" iconright="arrow-button-down-1" iconrightclickable="true" class="input-itself input size-md normal"><!----><span data-v-4ce83152="" data-v-400546b3="" class="unnnic-icon unnnic-icon__size--sm unnnic--clickable unnnic-icon-scheme--neutral-dark icon-right clickable"><svg viewBox="0 0 40 40" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M5.31471 14.0096L19.0432 29.5967C19.1676 29.7264 19.3169 29.8296 19.4822 29.9001C19.6474 29.9707 19.8253 30.007 20.005 30.007C20.1847 30.007 20.3625 29.9707 20.5278 29.9001C20.6931 29.8296 20.8424 29.7264 20.9668 29.5967L34.689 14.0096C34.7987 13.886 34.8825 13.7415 34.9353 13.5849C34.9882 13.4282 35.009 13.2625 34.9965 13.0977C34.984 12.9328 34.9385 12.7721 34.8627 12.6252C34.7869 12.4783 34.6824 12.3481 34.5553 12.2423L32.243 10.2963C31.9852 10.0789 31.6522 9.97163 31.316 9.99759C30.9797 10.0235 30.6671 10.1807 30.4457 10.435L20.2406 22.25C20.2112 22.2841 20.1749 22.3114 20.134 22.3302C20.0931 22.3489 20.0487 22.3587 20.0037 22.3587C19.9588 22.3587 19.9143 22.3489 19.8734 22.3302C19.8326 22.3114 19.7962 22.2841 19.7669 22.25L9.558 10.4338C9.33639 10.1806 9.02374 10.0252 8.68809 10.0013C8.52167 9.9865 8.35397 10.0052 8.19489 10.0563C8.03582 10.1074 7.8886 10.1898 7.76195 10.2988L5.44845 12.2423C5.32061 12.3479 5.21534 12.4781 5.13889 12.6252C5.06245 12.7723 5.0164 12.9332 5.0035 13.0985C4.99099 13.2635 5.01212 13.4294 5.06562 13.586C5.11911 13.7426 5.20386 13.8868 5.31471 14.0096Z" fill="#3B414D" class="primary"></path></svg></span></div>
                  <div data-v-6fe2671f="" class="dropdown-data" style="position: fixed; top: 0px;"></div></span>
                </div>
              </div>
              <div data-v-48c0c241="" class="vtex-modal__content__form__input__subdomain unnnic-form md">
                <p data-v-48c0c241="" class="unnnic-form__label">Subdomain</p>
                <div data-v-400546b3="" data-v-48c0c241="" class="unnnic-form-input text-input size--md" v-bind="[object Object]" mask=""><input data-v-97b44e8c="" data-v-400546b3="" type="text" size="0" v-bind="[object Object]" placeholder="Ex: shop.myvtex.com" class="input-itself input size-md normal">
                  <!---->
                  <!---->
                </div>
                <!---->
              </div>
              <div class="vtex-modal__content__form__keys">
                <div data-v-48c0c241="" class="vtex-modal__content__form__input__subdomain unnnic-form md">
                  <p data-v-48c0c241="" class="unnnic-form__label">AppKey</p>
                  <div data-v-400546b3="" data-v-48c0c241="" class="unnnic-form-input text-input size--md" v-bind="[object Object]" mask=""><input data-v-97b44e8c="" data-v-400546b3="" type="text" size="0" v-bind="[object Object]" placeholder="Insert your AppKey" class="input-itself input size-md normal">
                    <!---->
                    <!---->
                  </div>
                  <!---->
                </div>
                <div data-v-48c0c241="" class="vtex-modal__content__form__input__subdomain unnnic-form md">
                  <p data-v-48c0c241="" class="unnnic-form__label">AppToken</p>
                  <div data-v-400546b3="" data-v-48c0c241="" class="unnnic-form-input text-input size--md" v-bind="[object Object]" mask=""><input data-v-97b44e8c="" data-v-400546b3="" type="text" size="0" v-bind="[object Object]" placeholder="Insert your AppToken" class="input-itself input size-md normal">
                    <!---->
                    <!---->
                  </div>
                  <!---->
                </div>
              </div> <span class="vtex-modal__content__form__guide">
        Don't know your AppKey and AppToken?
        <a href="https://help.vtex.com/en/tutorial/application-keys--2iffYzlvvz4BDMr6WGUtet" target="_blank" rel="noopener noreferrer">
          Click here
        </a>
        to access the documentation on VTEX and find out.
      </span>
            </div>
          </div>
        </div>
      </div>
      <!---->
      <div data-v-4c459454="" class="unnnic-modal-container-background-button"><button data-v-2d9f2cbf="" class="unnnic-button unnnic-button--size-large unnnic-button--tertiary">
          <!---->
          <!---->
          <!----><span data-v-2d9f2cbf="" class="unnnic-button__label">
    Cancel
    </span>
          <!----></button><button data-v-2d9f2cbf="" class="unnnic-button unnnic-button--size-large unnnic-button--secondary">
          <!---->
          <!---->
          <!----><span data-v-2d9f2cbf="" class="unnnic-button__label">
    Continue
    </span>
          <!----></button></div>
    </div>
  </div>
  </div>
</transition-stub>
`;
